### ISSUE TITLE
Diffusers image-to-image with odd dimensions are broken

### ISSUE THREAD

**Issue by isidentical** [siliconflow/onediff#602]
Similar to #566 or #527, the uncommon dimensions are still broken for image to image and inpainting workloads. This is been tested on top of commit c07473d671f6775d2b5cef5d79d06b34ebb8543e (freshly built graph) and latest oneflow community edition. 

E.g. 528x528 image yields the following error:
```
2024-01-31 18:04:51.967 [stdout   ] [ERROR](GRAPH:OneflowGraph_0:OneflowGraph) run got error: <class 'oneflow._oneflow_internal.exception.Exception'> Check failed: (33 == 34) 
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/job/job_interpreter.cpp", line 307, in InterpretJob
2024-01-31 18:04:51.967 [stdout   ]     RunNormalOp(launch_context, launch_op, inputs)
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/job/job_interpreter.cpp", line 219, in RunNormalOp
2024-01-31 18:04:51.967 [stdout   ]     it.Apply(*op, inputs, &outputs, OpExprInterpContext(empty_attr_map, JUST(launch_op.device)))
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/framework/op_interpreter/eager_local_op_interpreter.cpp", line 84, in NaiveInterpret
2024-01-31 18:04:51.967 [stdout   ]     [&]() -> Maybe<const LocalTensorInferResult> { LocalTensorMetaInferArgs ... mut_local_tensor_infer_cache()->GetOrInfer(infer_args)); }()
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/framework/op_interpreter/eager_local_op_interpreter.cpp", line 87, in operator()
2024-01-31 18:04:51.967 [stdout   ]     user_op_expr.mut_local_tensor_infer_cache()->GetOrInfer(infer_args)
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/framework/local_tensor_infer_cache.cpp", line 210, in GetOrInfer
2024-01-31 18:04:51.967 [stdout   ]     Infer(*user_op_expr, infer_args)
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/framework/local_tensor_infer_cache.cpp", line 178, in Infer
2024-01-31 18:04:51.967 [stdout   ]     user_op_expr.InferPhysicalTensorDesc( infer_args.attrs ... ) -> TensorMeta* { return &output_mut_metas.at(i); })
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/core/framework/op_expr.cpp", line 602, in InferPhysicalTensorDesc
2024-01-31 18:04:51.967 [stdout   ]     physical_tensor_desc_infer_fn_(&infer_ctx)
2024-01-31 18:04:51.967 [stdout   ]   File "oneflow/user/ops/concat_op.cpp", line 55, in InferLogicalTensorDesc
2024-01-31 18:04:51.967 [stdout   ]     CHECK_EQ_OR_RETURN(in_desc.shape().At(i), out_dim_vec.at(i))
2024-01-31 18:04:51.967 [stdout   ] Error Type: oneflow.ErrorProto.check_failed_error
2024-01-31 18:04:51.967 [stderr   ] ERROR [2024-01-31 18:04:51] - Exception in __call__: e=Exception('Check failed: (33 == 34) \n  File "oneflow/core/job/job_interpreter.cpp", line 307, in InterpretJob\n    RunNormalOp(launch_context, launch_op, inputs)\n  File "oneflow/core/job/job_interpreter.cpp", line 219, in RunNormalOp\n    it.Apply(*op, inputs, &outputs, OpExprInterpContext(empty_attr_map, JUST(launch_op.device)))\n  File "oneflow/core/framework/op_interpreter/eager_local_op_interpreter.cpp", line 84, in NaiveInterpret\n    [&]() -> Maybe<const LocalTensorInferResult> { LocalTensorMetaInferArgs ... mut_local_tensor_infer_cache()->GetOrInfer(infer_args)); }()\n  File "oneflow/core/framework/op_interpreter/eager_local_op_interpreter.cpp", line 87, in operator()\n    user_op_expr.mut_local_tensor_infer_cache()->GetOrInfer(infer_args)\n  File "oneflow/core/framework/local_tensor_infer_cache.cpp", line 210, in GetOrInfer\n    Infer(*user_op_expr, infer_args)\n  File "oneflow/core/framework/local_tensor_infer_cache.cpp", line 178, in Infer\n    user_op_expr.InferPhysicalTensorDesc( infer_args.attrs ... ) -> TensorMeta* { return &output_mut_metas.at(i); })\n  File "oneflow/core/framework/op_expr.cpp", line 602, in InferPhysicalTensorDesc\n    physical_tensor_desc_infer_fn_(&infer_ctx)\n  File "oneflow/user/ops/concat_op.cpp", line 55, in InferLogicalTensorDesc\n    CHECK_EQ_OR_RETURN(in_desc.shape().At(i), out_dim_vec.at(i))\nError Type: oneflow.ErrorProto.check_failed_error')
```

**Comment by strint** [comment]
@isidentical  Is there a simple code to reproduce this?

**Comment by isidentical** [comment]
yeah, sure:
```py
import torch
from onediffx import compile_pipe
from PIL import Image
from onediffx.deep_cache import StableDiffusionXLPipeline

t2i_pipe = StableDiffusionXLPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.float16,
    variant="fp16",
)

t2i_pipe.to("cuda")
t2i_pipe = compile_pipe(t2i_pipe)

from diffusers import AutoPipelineForImage2Image

i2i_pipe = AutoPipelineForImage2Image.from_pipe(t2i_pipe)
i2i_pipe.to("cuda")

# Works
for dimensions in [(512, 512), (544, 544), (576, 578)]:
    i2i_pipe(
        prompt="a photo of a cat",
        negative_prompt="",
        image=Image.new("RGB", dimensions),
        num_inference_steps=20,
    )

# Doesn't work
i2i_pipe(
    prompt="a photo of a cat",
    negative_prompt="",
    image=Image.new("RGB", (528, 528)),
    num_inference_steps=20,
)
```

**Comment by lixiang007666** [comment]
You can merge this PR, retest:
https://github.com/siliconflow/onediff/pull/620
and later we will merge it into the main branch.

**Comment by isidentical** [comment]
@lixiang007666 does it fix #615 as well? Without it I dont think we can use your branch

**Comment by lixiang007666** [comment]
Of course~ we also need to merge https://github.com/siliconflow/onediff/pull/617. You can wait for the update to the main branch, and we will mention you.

**Comment by lixiang007666** [comment]
> @lixiang007666 does it fix #615 as well? Without it I dont think we can use your branch

Of course, we also need to merge https://github.com/siliconflow/onediff/pull/617. You can wait for the update to the main branch, and we will mention you."

**Comment by isidentical** [comment]
@lixiang007666 can confirm it works!
